### PR TITLE
make formula for solr@7.7

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Now(March 16th, 2019), we can install solr@8 and solr@6 by brew install,
but we cannot do solr@7.
So, I add a formula for solr@7

### CODE REVIEW

**Comment by SMillerDev** [comment]
Are all of these still supported upstream?

**Comment by hyuraku** [comment]
@SMillerDev 
[Official Solr site](http://lucene.apache.org/solr/community.html#support) says below about support
>Apache Solr is under active development with frequent feature releases on the current major version. The previous major version may still receive security- and bug fixes as point releases, thus taking the role of the LTS (Long Term Support) version. Older versions are considered EOL (End Of Life) and will not be further updated. 

I thought Solr@8 and Solr@7 are supported but Solr@6 isn't supported.

**Comment by SMillerDev** [comment]
Could you make this solr@7 then and remove solr@6?

**Comment by hyuraku** [comment]
@SMillerDev 
Sure, and I have a question.
Which should I remove solr@6 in this PR or in another one?

**Comment by SMillerDev** [comment]
I'd say this PR.

File: Formula/solr@7.7.rb [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
   homepage "https://lucene.apache.org/solr/"
-  url "https://www.apache.org/dyn/closer.cgi?path=lucene/solr/5.5.5/solr-5.5.5.tgz"
-  mirror "https://archive.apache.org/dist/lucene/solr/5.5.5/solr-5.5.5.tgz"
-  sha256 "2bbe3a55976f118c5d8c2382d4591257f6e2af779c08c6561e44afa3181a87c1"
+  url "https://www.apache.org/dyn/closer.cgi?path=lucene/solr/7.7.1/solr-7.7.1.tgz"
+  sha256 "4fb85f12af045b28f6cb935b2f3739d59ec61ad1288ffe44c7ede64e614f28c4"
 
   bottle :unneeded
 
-  keg_only :versioned_formula
```
Comment (fxcoudert) [inline Formula/solr@7.7.rb]: Please restore `keg_only :versioned_formula`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by fxcoudert — APPROVED** [review]

**Comment by gmkey** [comment]
The Apache Foundation has all versions prior to solr  6.4 are unsupported. So I guess solr@5.5 is obsolete, but Solr 6.6 isn’t!. Would it be possible to reinstate Solr 6.6?

Source:
http://lucene.apache.org/solr/community.html#support

**Comment by fxcoudert** [comment]
This appears inconsistent, though:

> Solr 8.0.0 is the most recent Apache Solr release

and

> The previous major version may still receive security- and bug fixes as point releases, thus taking the role of the LTS (Long Term Support) version. Older versions are considered EOL (End Of Life) and will not be further updated.

So it simply looks like the webpage has not been updated. Clarification would be welcome.

**Pull request closed without merging**